File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();
        int left = 0;
        int maxCount = 0;

        HashSet<Character> set = new HashSet<>();

        for(int right = 0; right < n; right++)
        {
            while(set.contains(s.charAt(right)))
            {
                set.remove(s.charAt(left));
                left++;
            }
            set.add(s.charAt(right));
            maxCount = Math.max(maxCount, right-left+1);
        }
        return maxCount;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> st;

        int left = 0;
        int maxl = INT_MIN;
        if(s.size()==0)return 0;

        for(int right = 0; right < s.size(); right++)
        {
            while(st.count(s[right]))
            {
                st.erase(s[left]);
                left++;
            }
            st.insert(s[right]);
            maxl = max(maxl, right - left + 1);
        }
        return maxl;
    }
};